rosbridge_server/src/rosbridge_server/tcp_handler.py

```diff
@@ -86,33 +91,37 @@ def recv_bson(self):
 
     def handle(self):
         """
-        Listen for TCP messages
+        Listen for TCP messages and do all the work to service a request.
         """
         cls = self.__class__
         self.request.settimeout(cls.socket_timeout)
-        while 1:
+        while True:
             try:
-              if self.bson_only_mode:
-                  if self.recv_bson() == None:
-                      break
-                  continue
-
-              # non-BSON handling
-              data = self.request.recv(cls.incoming_buffer)
-              # Exit on empty string
-              if data.strip() == '':
-                  break
-              elif len(data.strip()) > 0:
-                  self.protocol.incoming(data.decode().strip(''))
-              else:
-                  pass
+                if self.bson_only_mode: # is False
```

Comment:
what is the `is False` comment referring to?

Reply:
That was just a little memory aid that I forgot to remove.
Thanks for the hint and the fast response. I removed in the newest commit.